specification/attestation.adoc

```diff
@@ -1,309 +1,885 @@
-the current layer (e.g. TSM-driver). The DICE flow outputs the CDI values 
-and the generated certificate; the private key associated with the 
-certificate may be optionally passed along with the CDI values to avoid the 
-need for re-derivation by the target layer. The UDS-derived public key is 
-certified by an external authority during manufacturing to root the 
-certificate chain in a HW RoT. 
-
-As a tangible example, the CDI private key for the TSM were used to sign a 
-leaf certificate for an attestation key for the TVM, the certificate chain 
-may look like this:
-
-image:img_6.png[]
-Figure 6: Attestation Certificate generation
-
-This attestation certificate can be used in a challenge/response protocol 
-to a remote relying party which must verify the certificate chain for the 
-attestation key used to sign the relying party challenge. 
-
-The Attestation key and certificate generation for TVMs may be performed 
-with a U-mode TSM component called the Owner Signing Authority Module 
-(OSAM) to enable a extension of the TCB to support interruptible signing 
-operations. The OSAM may execute as part of the TSM or may be executed in 
-the TSM U-mode to allow for the interruptibility models discussed in the 
-TSM operation section of this document.
-
-*TVM* *Attestation:*
-
-X.509 CDI Certificates are used to enable Attestation certificates derived 
-from the TSM CDI for each TVM hosted on the platform. All standard fields 
-of a CDI certificate are described in the following table. This certificate 
-can be generated given a CDI_Public key and the DICE input values.
-
+.AP-TEE Workload TCB Elements
+[cols="^1, ^5v, ^4v", options="header"]
 |===
-| *Field*  | *Description* 
-| signatureAlgorithm | id-ecdsa-with-SHA256 per RFC 5758 recommended. 
-Other signatureAlgorithms may be used.
-| signatureValue | 64 byte ECDSA signature, using UDS_Private or a previous 
-CDI_Private as the signing key
-| version | v3
-| serialNumber | CDI_ID in ASN.1 INTEGER form
-| signature | id-ecdsa-with-SHA256 per RFC 5758
-| issuer | “<UDS_ID> or <CDI_ID>” UDS_ID, CD_ID are hex encoded lower case
-| validity | The validity values are populated as follows: notBefore can be 
-any time known to be in the past, and notAfter is set to the standard 
-value used to indicate no well-known expiry date, “99991231235959Z” per 
-RFC 5280.  
-| subject | “<CDI_ID>” where CDI_ID is hex encoded lower case
-| subjectPublicKeyInfo | When using ECDSA, per RFC 5480 (id-ecPublicKey)
-| issuerUniqueID | Not used
-| subjectUniqueID | Not used
-| extensions | Standard extensions are included as well as a custom TCG 
-extension which holds information about the measurements used to derive 
-CDI values. Both are described 
-below.                                                  
+|TCB |Hardware Elements |Software Elements
+
+.4+.^|Platform
+    <| HW RoT for boot, measurement and storage
+ .4+<| All M-mode firmwares, except for the TSM-driver
```

Comment:
do you mean - including the TSM-driver?

Reply:
I mean to exclude the TSM-driver from the platform's TCB. I look at the platform TCB as containing all TCB elements that are not AP-TEE specific, which excludes the TSM-Driver.

Now, thinking a bit more about it: In the context of this specification (AP-TEE), the platform *will* have a `TSM-driver`, so I guess we should add it to the platform TCB and not the TSM one as even non TSM/TVM elements could be influenced by this M-mode component. I fixed it.